components/usage/pkg/db/team_membership_test.go

```diff
@@ -30,3 +31,70 @@ func TestTeamMembership_WriteRead(t *testing.T) {
 	require.NoError(t, tx.Error)
 	require.Equal(t, membership, read)
 }
+
+func TestListTeamMembershipsForUserIDs(t *testing.T) {
+	conn := db.ConnectForTests(t)
+	userWithTwoTeams := uuid.New()
+	memberships := []db.TeamMembership{
+		{
+			ID:     uuid.New(),
+			TeamID: uuid.New(),
+			UserID: uuid.New(),
+			Role:   db.TeamMembershipRole_Member,
+		},
+		{
+			ID:     uuid.New(),
+			TeamID: uuid.New(),
+			UserID: userWithTwoTeams,
+			Role:   db.TeamMembershipRole_Member,
+		},
+		{
+			ID:     uuid.New(),
+			TeamID: uuid.New(),
+			UserID: userWithTwoTeams,
+			Role:   db.TeamMembershipRole_Member,
+		},
+	}
+
+	tx := conn.Create(&memberships)
+	require.NoError(t, tx.Error)
+
+	for _, scenario := range []struct {
+		Name     string
+		UserIDs  []uuid.UUID
+		Expected int
+	}{
+		{
+			Name:     "no user ids return empty",
+			UserIDs:  nil,
+			Expected: 0,
+		},
+		{
+			Name:     "no matching user IDs returns empty",
+			UserIDs:  []uuid.UUID{uuid.New()},
+			Expected: 0,
```

Comment:
Over-nit: This test has a non-zero chance of misfiring. 😂

Reply:
> UUIDs have 122 bits of entropy so the chance of two random UUIDs colliding is about 10^-37.

> If you generate 2^46 UUIDs (approximately 1 petabyte of entropy) the chance of getting a collision is 1 in 50 billion.[1] You're 170 times more likely to win the jackpot in Powerball.[2]

https://www.quora.com/Has-there-ever-been-a-UUID-collision